docs/src/transforms.md

```diff
@@ -10,77 +11,199 @@ DocTestSetup = quote
+
+Transforms can be applied to `AbstractArray` data using a `dims` keyword argument.
+This will apply the `Transform` to slices of the array along dimensions determined by `dims`.
+For example, given a `Matrix`, `dims=1` applies to each column, and `dims=2` applies
+to each row.
+
+!!! note
+
+  In general, the `dims` argument uses the convention of `mapslices`, which is called behind the scenes when applying transforms to slices of data.
+  In practice, this means that users can expect the `dims` keyword to behave exactly as `mean(A; dims=d)` would; the transformation will be applied to the elements along the dimension `d` and, for operations like `mean` or `sum`, reduce across this dimension.
+
+```jldoctest transforms
+julia> M
+3×2 Array{Int64,2}:
+ 2  4
+ 1  5
+ 3  6
+
+julia> normalize_cols = MeanStdScaling(M; dims=1);
+
+julia> normalize_cols(M; dims=1)
 3×2 Array{Float64,2}:
   0.0  -1.0
  -1.0   0.0
   1.0   1.0
+
+julia> normalize_rows = MeanStdScaling(M; dims=2);
+
+julia> normalize_rows(M; dims=2)
+3×2 Array{Float64,2}:
+ -0.707107  0.707107
+ -0.707107  0.707107
+ -0.707107  0.707107
 ```
 
-Note that some `Transform`s have restrictions on how they can be applied once constructed. For instance, `MeanStdScaling` stores the mean and standard deviation of some data for specified dimensions (for arrays) or columns (for tables). So `MeanStdScaling` should only be applied to the same data type and along the same dimensions or subset of columns specified in construction.
+### Using `dims` and `inds` together
+
+When using `dims` with `inds`, the `inds` change from being the global indices of the array to the relative indices of each slice.
+For example, the following is another way to square the second column of an array, applying to  index 2 of each row:
+
+```jldoctest transforms
+julia> FeatureTransforms.apply(M, p; dims=2, inds=[2])
+3×1 Array{Int64,2}:
+ 16
+ 25
+ 36
+```
+
+## Applying to `Table`
+
+### Default
 
-We can also provide the `inds` keyword to apply the `Transform` to certain indices along the
-array slices. For example, to only scale every odd row:
+Without specifying optional arguments, a `Transform` is applied to every column of a `Table` independently:
 
 ```jldoctest transforms
-julia> FeatureTransforms.apply(M, scaling; dims=1, inds=1:2:size(M, 1))
-2×2 Array{Float64,2}:
- 0.0  -1.0
- 1.0   1.0
+julia> nt = (a = [2, 1, 3], b = [4, 5, 6]);
+
+julia> scaling = MeanStdScaling(nt);
+
+julia> FeatureTransforms.apply!(nt, scaling)
+(a = [0, -1, 1], b = [-1, 0, 1])
 ```
 
-### `Table` data
+!!! note
+
+  The non-mutating `apply` method for `Table` data returns a `Vector` of `Vector`s, one for each column, e.g.
+
+  ```julia-repl
+  julia> FeatureTransforms.apply(nt, scaling)
+  2-element Array{Array{Float64,1},1}:
+   [-2.0, -3.0, -1.0]
+   [-6.0, -5.0, -4.0]
+  ```
+
+  This is so users are free to decide what to name the results of the transformation, whether to append to the original table, etc.
```

Comment:
Would be good to put this directly under the sentence above. I couldn't remember why we did this but knew there was a reason.